pkg/generator_test.go

```diff
@@ -2433,6 +2433,26 @@ func NewMockRequesterGenerics[TAny interface{}, TComparable comparable, TSigned
 	s.checkGeneration("generic.go", "RequesterGenerics", true, "", expected)
 }
 
+func (s *GeneratorSuite) TestGetLocalizedPathVendored() {
+	generator := s.getGenerator(testFile, "Requester", false, "")
+	result := generator.getLocalizedPath(context.Background(), "vendor/github.com/nats-io/nats.go/js.go")
+
+	s.Equal(
+		"github.com/nats-io/nats.go", result,
```

Comment:
If you're vendoring a directory don't you want the returned path to be on-disk path instead of the URL?

Reply:
I'm golang code, you put imports to a vendored directory in same way as from modules, that's why this current logic is in place.

Reply:
I see, I've never personally used vendored dependencies.

Reply:
Yeah, I used to. It's just a directory with all project dependencies in a form of a source code. In any case, we ought to check if a path element is a directory or not and if it's a file, remove it from path to build correct import statement.